src/library/scala/collection/Iterator.scala

```diff
@@ -74,7 +74,20 @@ import scala.annotation.unchecked.uncheckedVariance
   *  @define coll iterator
   */
 trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Iterator[A]] { self =>
+
+  /** Check if there is a next element available.
+    *
+    * @return `true` if there is a next element, `false` otherwise
+    * @note   Reuse: $preservesIterator
+    */
   def hasNext: Boolean
+
+  /** Return the next element and advance the iterator.
+    *
+    * @throws NoSuchElementException if there is no next element.
+    * @return the next element.
+    * @note   Reuse: $preservesIterator
```

Comment:
I’m not sure we should add this note because the `next()` method doesn’t really preserves the underlying iterator since it consumes one element.

Reply:
The `preservesIterator` definition above talks about whether or not you can continue to use the iterator, not about whether or not the iterator is modified. Maybe the definition should be adjusted, or another one added, to account for the iterator being modified but still valid for further use?

Reply:
I think that instead of using `$preservesIterator` we should say that the iterator is modified but still valid for further use, as you said. (No need to create a `@define` for that since it’s the only occurrence of that note)